packages/fxa-settings/src/models/Account.ts

```diff
@@ -33,31 +62,9 @@ export const GET_ACCOUNT = gql`
   }
 `;
 
-export interface AccountData {
-  uid: string;
-  displayName: string | null;
-  avatarUrl: string | null;
-  accountCreated: number;
-  passwordCreated: number;
-  recoveryKey: boolean;
-  emails: {
-    email: string;
-    isPrimary: boolean;
-    verified: boolean;
-  }[];
-  attachedClients: {
-    clientId: string;
-    isCurrentSession: boolean;
-    userAgent: string;
-    deviceType: string;
-    deviceId: string;
-  }[];
-  totp: {
-    exists: boolean;
-    verified: boolean;
-  };
-  subscriptions: {
-    created: number;
-    productName: string;
-  }[];
+export function useAccount() {
+  const { data } = useQuery<{ account: Account }>(GET_ACCOUNT, {
+    fetchPolicy: 'cache-only',
+  });
+  return data!.account;
```

Comment:
If I'm understanding the docs correctly, `loading` would never be used because `cache-only`, but it looks like this might still need to handle (or maybe just return) `error`. Something like:

```typescript
const { data, error } = useQuery<{ account: Account }>(GET_ACCOUNT, {
  fetchPolicy: 'cache-only',
});

return { account: data!.account, error };
```

Reply:
Err, unless `error` is only ever possible when performing network requests. I'm unsure if an error can occur after the initial fetch.

Reply:
Yeah, I want to look at the source to figure out if `error` can occur with cache lookups

Reply:
From https://github.com/apollographql/apollo-client/blob/master/src/core/ObservableQuery.ts#L128 it'll return the same result as the original `GET_INITIAL_STATE` query. If there was an `error` then, there will be an `error` here, but since we already handle that in the `App` component it's safe to assume if we get this far there is no `error`.